### PR TITLE
Fix STABmons Move Legality

### PR DESCRIPTION
**Pull request by KrisXV**

### CODE REVIEW

File: data/rulesets.ts [hunk 1]
```diff
@@ -927,6 +928,12 @@ export const BattleFormats: {[k: string]: FormatsData} = {
...
 						for (const formeid of baseSpecies.otherFormes) {
+							let skipForme = false;
+							if (baseSpecies.prevo) {
+								const prevo = dex.getSpecies(baseSpecies.prevo);
+								skipForme = prevo.evos.map(toID).includes(toID(formeid));
+							}
+							if (skipForme) continue;
 							const forme = dex.getSpecies(formeid);
 							if (forme.changesFrom === originalForme.name) {
 								types = types.concat(forme.types);
```
Comment (Zarel) [inline data/rulesets.ts:939]:
```suggestion
						for (const formeName of baseSpecies.otherFormes) {
							if (baseSpecies.prevo) {
								const prevo = dex.getSpecies(baseSpecies.prevo);
								if (prevo.evos.includes(formeName)) continue;
							}
							const forme = dex.getSpecies(formeName);
							if (forme.changesFrom === originalForme.name) {
								types = types.concat(forme.types);
```

The `toID` mapping seems excessive...

Comment (KrisXV) [reply data/rulesets.ts:939]: I disagree. It's there to ensure that typos (such as a miscapitalized O) in the data file don't get in the way of validation.

Comment (KrisXV) [reply data/rulesets.ts]: In any case, I did this (locally) and pushed because the type of `Species#evos` was incorrect.

File: sim/dex-data.ts [hunk 2]
```diff
@@ -587,9 +587,9 @@ export class Species extends BasicEffect implements Readonly<BasicEffect & Speci
 	/** Added type (used in OMs). */
 	readonly addedType?: string;
 	/** Pre-evolution. '' if nothing evolves into this Pokemon. */
-	readonly prevo: ID;
+	readonly prevo: string;
```
Comment (Zarel) [inline sim/dex-data.ts:590]: Actually wait, did we switch these over?

Comment (Zarel) [reply sim/dex-data.ts:590]: Looks like yes; whew.

File: data/rulesets.ts (hunk 1 above)
Comment (Zarel) [reply data/rulesets.ts]: > It's there to ensure that typos (such as a miscapitalized O) in the data file don't get in the way of validation.

This is why I added data tests, so we wouldn't have to worry about stuff like this...

**Comment by Zarel** [comment]
Thanks!

File: data/rulesets.ts (hunk 1 above)
Comment (KrisXV) [reply data/rulesets.ts]: Oh yeah

**Pull request merged**